Assets/Features/UI Design/Scripts/Runtime/Utils/LapTime.cs

```diff
@@ -10,26 +12,63 @@ public class LapTime : IEnumerable<float>
         public float Sum => times[SumIndex];
         public readonly int sectorCount;
         public int TimesCount { get; }
+        public bool IsCompleted => pointer == sectorCount;
 
-        public LapTime(float[] s)
+        private int pointer;
+        public LapTime(int sectorCount, float[] sectorTimes)
         {
-            sectorCount = s.Length;
+            if (sectorTimes.Length > sectorCount)
+            {
+                string msg = $"The provider array lenght is bigger than sector count. {sectorTimes.Length} > {sectorCount}";
+                throw new ArgumentException(msg);
+            }
+
+            this.sectorCount = sectorCount;
             this.times = new float[sectorCount + 1];
 
             int i;
             float sum = 0f;
-            for (i = 0; i < sectorCount; i++)
+
+            for (i = 0; i < sectorTimes.Length; i++)
             {
-                sum += s[i];
-                times[i] = s[i];
+                sum += sectorTimes[i];
+                times[i] = sectorTimes[i];
             }
+            pointer = i;
+            for (i = sectorTimes.Length; i < sectorCount; i++)
+            {
+                times[i] = Mathf.Infinity;
```

Comment:
Here, the not raced sectors are marked to infinity. This way, the algorithms are transparent because an infinity time never is going to be selected